[llama-mm] Add export friendly TiledTokenPositionalEmbedding

larryliu0820 opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #6807
* __->__ #6806

Summary: As titled. This is used by CLIP vision encoder.

Test Plan: Rely on newly added unit tests.

Reviewers:

Subscribers:

Tasks:

Tags:

larryliu0820 committed:
[llama-mm] Add export friendly TiledTokenPositionalEmbedding

Summary: As titled. This is used by CLIP vision encoder.

Test Plan: Rely on newly added unit tests.

Reviewers:

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

angelayi commented on extension/llm/modules/test/test_position_embeddings.py:
```diff
@@ -116,3 +122,102 @@ def forward(self, x, aspect_ratio):
+        assert_close(y, ref_y)
+
+    def test_tiled_token_positional_embedding_aoti(self):
+        so = torch._export.aot_compile(
+            self.tpe,
+            args=(self.x, self.aspect_ratio),
+            options={"aot_inductor.package": True},
+            dynamic_shapes=(
+                self.dynamic_shape,
+                None,
+            ),  # assuming aspect ratio is static
+        )
+        with tempfile.TemporaryDirectory() as tmpdir:
+            path = package_aoti(os.path.join(tmpdir, "tpe.pt2"), so)
```
is it possible to rewrite this into using `torch.export.export` + `torch._inductor.aoti_compile_and_package`

larryliu0820 committed:
Update on "[llama-mm] Add export friendly TiledTokenPositionalEmbedding"


Summary: As titled. This is used by CLIP vision encoder.

Test Plan: Rely on newly added unit tests.

Reviewers:

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]
